nemo/collections/nlp/modules/common/megatron/megatron_utils.py

```diff
@@ -32,7 +32,7 @@
 
 MEGATRON_CACHE = os.path.join(os.path.dirname(str(TRANSFORMERS_CACHE)), 'megatron')
 
-CONFIGS = {'345m': {"hidden-size": 1024, "num-attention-heads": 16, "num-layers": 24, "max-seq-length": 512}}
+CONFIGS = {'345m': {"hidden_size": 1024, "num_attention_heads": 16, "num_layers": 24, "max_position_embeddings": 512}}
```

Comment:
These may cause issues with config files released with the Megatron checkpoints

Reply:
This is the format Megatron consumes. Where would we get different format from ?

Reply:
from Megatron models on NGC, for example, https://ngc.nvidia.com/catalog/models/nvidia:megatron_bert_345m/files

Reply:
Not sure if those are connected - we are just, like, supplying command line arguments and they are expected in this format?

Reply:
@borisfom Could you please add a Jenkins test when Megatron is initialized with the provided config.json and checkpoint, not loaded automatically - maybe replace this cased Megatron test with this one, to make sure nothing is broken with these changes.

Reply:
Yes please use the same files just initialize megatron using model.language_model.pretrained_model_name=megatron-bert-cased/uncased (notice 345m is missing - it's for non default models, like biomegatron) model.language_model.bert_checkpoint=path_to/model_optim_rng.pt model.language_model.bert_config=path_to/345m_config.json. In the current implementation, only default models are supported, init from provided config.json and chkp is broken because in the get_megatron_lm_model the default config/chkp is pulled.

Reply:
added test and fix https://github.com/NVIDIA/NeMo/pull/999